internal/storage/mongodb/trust_base.go

```diff
@@ -0,0 +1,152 @@
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/unicitynetwork/bft-go-base/types"
+)
+
+const trustBaseCollection = "trust_bases"
+
+var (
+	ErrTrustBaseNotFound      = errors.New("trust base not found")
+	ErrTrustBaseAlreadyExists = errors.New("trust base already exists")
+)
+
+type TrustBaseStorage struct {
+	collection *mongo.Collection
+}
+
+func NewTrustBaseStorage(db *mongo.Database) *TrustBaseStorage {
+	return &TrustBaseStorage{
+		collection: db.Collection(trustBaseCollection),
+	}
+}
+
+// Store verifies and stores a RootTrustBase.
+func (s *TrustBaseStorage) Store(ctx context.Context, trustBase types.RootTrustBase) error {
+	epoch := trustBase.GetEpoch()
+	version := uint64(trustBase.GetVersion())
+	mappedVersion := s.GetVersion(epoch)
+	if version != mappedVersion {
+		return fmt.Errorf("trust base version mismatch: got %d expected %d", version, mappedVersion)
+	}
+
+	// check if a trust base for this epoch already exists
+	existing, err := s.GetByEpoch(ctx, epoch)
+	if err != nil && !errors.Is(err, ErrTrustBaseNotFound) {
+		return fmt.Errorf("failed to check for existing trust base: %w", err)
+	}
+	if existing != nil {
+		return ErrTrustBaseAlreadyExists
+	}
+
+	// verify trust base extends previous trust base
+	var previousTrustBaseV1 *types.RootTrustBaseV1
+	if epoch > 0 {
+		previousTrustBase, err := s.GetByEpoch(ctx, epoch-1)
+		if err != nil {
+			return fmt.Errorf("previous trust base not found for epoch %d: %w", epoch-1, err)
+		}
+		var ok bool
+		previousTrustBaseV1, ok = previousTrustBase.(*types.RootTrustBaseV1)
+		if !ok {
+			return fmt.Errorf("failed to cast previous trust base to version 1 for epoch %d", epoch)
+		}
+	}
+	trustBaseV1, ok := trustBase.(*types.RootTrustBaseV1)
+	if !ok {
+		return fmt.Errorf("failed to cast provided trust base to version 1 for epoch %d", epoch)
+	}
+	if err := trustBaseV1.Verify(previousTrustBaseV1); err != nil {
```

Comment:
looked into `Verify()`, it checks `PreviousEntryHash`, but does not verify that new trustbase was signed by the previous majority set of validators. This seems like a security issue, when anyone could extend the tb and submit it a new one